Assets/Scripts/Pathfinding/PathfindingGridManager.cs

```diff
@@ -32,57 +39,85 @@ private void SetPathfindingGridManager(LogicGrid logicGrid)
                     }
                 }
             }
         }
 
         //Add neighbours to each pathnode
-        foreach(PathNode pathnode in m_pathNodeList)
+        foreach(PathNode pathnode in m_pathNodes)
         {
-            if (pathnode.X - 1 >= 0)
+            if (pathnode != null)
             {
-                // Left
-                pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y));
-                // Left Down
-                if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y - 1));
-                // Left Up
-                if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y + 1));
-            }
+                if (pathnode.X - 1 >= 0)
+                {
+                    // Left
+                    pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y));
+                    // Left Down
+                    if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y - 1));
+                    // Left Up
+                    if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X - 1, pathnode.Y + 1));
+                }
 
-            if (pathnode.X + 1 < m_logicGrid.Width)
-            {
-                // Right
-                pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y));
-                // Right Down
-                if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y - 1));
-                // Right Up
-                if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y + 1));
-            }
+                if (pathnode.X + 1 < m_logicGrid.Width)
+                {
+                    // Right
+                    pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y));
+                    // Right Down
+                    if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y - 1));
+                    // Right Up
+                    if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X + 1, pathnode.Y + 1));
+                }
 
-            // Down
-            if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X, pathnode.Y - 1));
-            // Up
-            if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X, pathnode.Y + 1));
+                // Down
+                if (pathnode.Y - 1 >= 0) pathnode.AddNeighbour(GetNode(pathnode.X, pathnode.Y - 1));
+                // Up
+                if (pathnode.Y + 1 < m_logicGrid.Height) pathnode.AddNeighbour(GetNode(pathnode.X, pathnode.Y + 1));
 
+            }
         }
 
-        m_pathfinding = new Pathfinding(m_pathNodeList);
+        m_pathfinding = new Pathfinding(pathNodeList);
+
+        SetUpPathTesting();
+    }
+
+    private void SetUpPathTesting()
```

Comment:
Le PathfindingManager initialise son propre test et c'est pt pas idéal pour de la prod, parce qu'on veut pas avoir ça dans le build serveur.
Idéalement, le PathfindingManager serait pas au courant de l'existence du PathTesting. 
Ce qu'on pourrait faire pour contourner le problème de l'ordre des appels (Awake, Start, etc) c'est 
- initialiser le pathtesting manuellement, par exemple avec un bouton comme on l'a fait pour Test Random Path
- créer un delegate dans PathfindingManager, delegate qui est appelée une fois le pathfinding initialisé et ensuite y faire subscribe le pathtesting
- une autre solution àlaquelle  je n'ai pas pensé

Bref l'important c'est d'ôter le setup du path testing du pathfinding manager parce qu'il va falloir le faire inévitablement quand on va vouloir builder pour de la prod

Reply:
Créer une entité dans l'editeur et lui mettre en parametre le manager?

Reply:
yes